test.js

```diff
@@ -391,3 +391,23 @@ it('missing source maps should also be cached', function(done) {
     '1', // The retrieval should only be attempted once
   ]);
 });
+
+it('finds source maps with charset specified', function() {
+  var sourceMap = createMultiLineSourceMap()
+  var source = [ 'throw new Error("test");' ];
+  var expected = [
+    'Error: test',
+    /^    at Object\.exports\.test \(.*\/line1\.js:1001:101\)$/
+  ];
+
+  fs.writeFileSync('.generated.js', 'exports.test = function() {' +
+    source.join('\n') + '};//@ sourceMappingURL=data:application/json;charset=utf8;base64,' +
+    new Buffer(sourceMap.toString()).toString('base64'));
+  try {
+    delete require.cache[require.resolve('./.generated')];
+    require('./.generated').test();
+  } catch (e) {
+    compareLines(e.stack.split('\n'), expected);
+  }
+  fs.unlinkSync('.generated.js');
```

Comment:
`compareStackTrance()` is hard coded to use `data:application/json;base64,` (aka. no charset). I didn't want to change it because we should test for both. I didn't want to add an extra parameter because I felt that it added too much clutter vs. the benefit, thus I just inlined the relevant parts in this test.